core/src/main/java/org/infinispan/notifications/cachelistener/NotifyHelper.java

```diff
@@ -14,54 +17,73 @@
    public static void entryCommitted(CacheNotifier notifier, FunctionalNotifier functionalNotifier,
                                      boolean created, boolean removed, boolean expired, CacheEntry entry,
                                      InvocationContext ctx, FlagAffectedCommand command, Object previousValue, Metadata previousMetadata) {
-      boolean isWriteOnly = (command instanceof WriteCommand) && ((WriteCommand) command).isWriteOnly();
-      if (removed) {
-         if (command instanceof RemoveCommand) {
-            ((RemoveCommand) command).notify(ctx, previousValue, previousMetadata, false);
-         } else if (command instanceof InvalidateCommand) {
-            notifier.notifyCacheEntryInvalidated(entry.getKey(), entry.getValue(), entry.getMetadata(), false, ctx, command);
-         } else {
-            if (expired) {
-               notifier.notifyCacheEntryExpired(entry.getKey(), previousValue, previousMetadata, ctx);
+      // We only notify if there is no state transfer flag
+      if (extractStateTransferFlag(ctx, command) == null) {
```

Comment:
I had thought about this, but I tend to not like multiple locations in a method that can return. Thinking on it more it seems harmless enough, so it should be fine.